internal/spokes/spokes.go

```diff
@@ -88,18 +89,33 @@ func Exec(ctx context.Context, stdin io.Reader, stdout io.Writer, stderr io.Writ
 		config:           config,
 		statelessRPC:     *statelessRPC,
 		advertiseRefs:    *httpBackendInfoRefs,
-		quarantineFolder: filepath.Join(repoPath, "objects", quarantineID),
+		quarantineFolder: quarantineFolder,
 		governor:         g,
 	}
 
+	registerShutdownHooks(rp)
+
 	if err := rp.execute(ctx); err != nil {
 		g.SetError(1, err.Error())
+		rp.Close()
 		return 1, fmt.Errorf("unexpected error running spokes receive pack: %w", err)
 	}
 
 	return 0, nil
 }
 
+func registerShutdownHooks(hooks ...io.Closer) {
```

Comment:
Using `io.Closer` here implies that this should be used with IO handles, when really it's just running a func. Since it's also not checking the returned error, would it make sense to change the hooks' type to `func()`? This is what [`testing.T.Cleanup`](https://pkg.go.dev/testing#T.Cleanup) does, and it seems like a good pattern to use here.

```suggestion
func registerShutdownHooks(hooks ...func()) {
```

If you make this change, you'll also need to change the registration to be something like `registerShutdownHooks(rp.Close)` or `registerShutdownHooks(rp.OnError)`.